src/NuGet.Core/NuGet.ProjectModel/ProjectLockFile/PackagesLockFileUtilities.cs

```diff
@@ -471,7 +471,7 @@ private static (bool, string) HasP2PDependencyChanged(IEnumerable<LibraryDepende
             {
                 var matchedP2PLibrary = projectDependency.Dependencies.FirstOrDefault(dep => StringComparer.OrdinalIgnoreCase.Equals(dep.Id, dependency.Name));
 
-                if (matchedP2PLibrary == null || !EqualityUtility.EqualsWithNullCheck(matchedP2PLibrary.VersionRange, dependency.LibraryRange.VersionRange))
```

Comment:
+1 to Erick's suggestion. In the PR template, you selected "N/A" under tests, but you didn't explain why adding tests is not applicable.  Since existing tests did not catch this bug, we should add a new test to minimize risk of regression in the future.

Reply:
I'm not confident this is the correct change either. 

The full range affects the restore algorithm, not just the minimum version. 

Shouldn't we be writing the full range in the lock file instead? 

Note that might be a format breaking change.